web/src/stream_edit_subscribers.js

```diff
@@ -271,6 +273,27 @@ function remove_subscriber({stream_id, target_user_id, $list_entry}) {
         return;
     }
 
+    if (sub.invite_only && peer_data.get_subscriber_count(stream_id) === 1) {
+        const stream_name_with_privacy_symbol_html = render_inline_decorated_stream_name({
+            stream: sub,
+        });
+
+        const html_body = render_unsubscribe_private_stream_user_modal();
```

Comment:
We can use `render_unsubscribe_private_stream_modal` here but we would have to pass `display_stream_archive_warning` as `false` as we want a different message for archive warning which will be passed in the message field itself and might confuse users. So, this seems good to me for now and we can change it later if feedback is different from Tim.